src/NHibernate/Loader/Criteria/CriteriaQueryTranslator.cs

```diff
@@ -41,10 +41,11 @@ public class EntityJoinInfo
 		private readonly IDictionary<String, ICriteriaInfoProvider> nameCriteriaInfoMap =
 			new Dictionary<string, ICriteriaInfoProvider>();
 
+		private readonly HashSet<ICollectionPersister> uncacheableCollectionPersisters = new HashSet<ICollectionPersister>();
 		private readonly ISet<ICollectionPersister> criteriaCollectionPersisters = new HashSet<ICollectionPersister>();
 		private readonly IDictionary<ICriteria, string> criteriaSQLAliasMap = new Dictionary<ICriteria, string>();
 		private readonly IDictionary<string, ICriteria> aliasCriteriaMap = new Dictionary<string, ICriteria>();
-		private readonly IDictionary<string, ICriteria> associationPathCriteriaMap = new LinkedHashMap<string, ICriteria>();
+		private readonly Dictionary<string, CriteriaImpl.Subcriteria> associationPathCriteriaMap = new Dictionary<string, CriteriaImpl.Subcriteria>();
```

Comment:
`LinkedHashMap` guarantees the dictionary elements will be iterated in the same order they were added. This is not the case of `Dictionary`, which, with current implementations, will have a different iteration order as soon as some elements were removed.

So this change is not neutral. But it looks like preserving the order has no purpose here, so this is ok. (Similarly `associationPathJoinTypesMap` does not need to be a `LinkedHashMap`.)